astropy/coordinates/tests/test_regression.py

```diff
@@ -751,3 +751,70 @@ def test_regression_10291():
     assert_quantity_allclose(
         venus.separation(sun), 554.427 * u.arcsecond, atol=0.001 * u.arcsecond
     )
+
+
+@pytest.mark.parametrize("coord_cls", [ICRS, SkyCoord])
+class TestRegression16998:
+    """Direct tests of the underlying problem causing gh-16998.
+
+    Note that the issue itself was of columns missing in data written to a file.
+    That is now tested directly by the "icrs" column in
+    astropy/io/tests/mixin_columns.py.
+    Here, we test the underlying problem, that .info._represent_as_dict()
+    did not return a full set of columns in some cases.
+    """
+
+    @classmethod
+    def setup_class(self):
+        self.ra = [0, 10] * u.deg
+        self.dec = [20, 30] * u.deg
+        self.pm_ra_cosdec = [40, 50] * u.mas / u.yr
+        self.pm_dec = [60, 70] * u.mas / u.yr
+        self.radial_velocity = [80, 90] * u.km / u.s
+
+    def get_kwargs(self, *args):
+        return {a: getattr(self, a) for a in args}
```

Comment:
The `get_kwargs()` helper method is too much complexity for a unit test. I'd prefer to store similar parameters together in dictionaries, which can be conveniently combined using the `|` operator. For example, see the definitions in https://github.com/astropy/astropy/blob/b266a4041ad4ebecbe1e07b329f70e03b71759e1/astropy/coordinates/tests/test_frames_with_velocity.py#L14-L17
and the usage throughout the file.

If these dictionaries are defined in the module scope then they can be reused in all the tests, but refactoring the already existing tests is beyond the scope of this pull request.

Reply:
I ended up doing it via parametrization, as you suggest below.